masq_lib/src/percentage.rs

```diff
@@ -1,77 +1,226 @@
 // Copyright (c) 2024, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
 
-use num::integer::mod_floor;
 use num::CheckedAdd;
 use num::CheckedSub;
 use num::{CheckedDiv, CheckedMul, Integer};
 use std::any::type_name;
 use std::fmt::Debug;
-use std::ops::Div;
-use std::ops::Mul;
+use std::ops::Rem;
+// Designed to store values from 0 to 100 and offer a set of handy methods for PurePercentage
+// operations over a wide variety of integer types. It is also to look after the least significant
+// digit on the resulted number in order to avoid the effect of a loss on precision that genuinely
+// comes with division on integers if a remainder is left over. The percents are always represented
+// by an unsigned integer. On the contrary, the numbers that it is applied on can take on both
+// positive and negative values.
 
-// It's designed for a storage of values from 0 to 100, after which it can be used to compute
-// the corresponding portion of many integer types. It should also take care of the least significant
-// digit in order to diminish the effect of a precision loss genuinly implied by this kind of math
-// operations done on integers.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct PurePercentage {
+    degree: u8,
+}
```

Comment:
It would be better if we keep the implementations of `PurePercentage` together with the struct. So, feel free to move things around in the way you seem fit.